Prog/Demographic trends.sas

```diff
@@ -0,0 +1,1036 @@
+
+		%Hud_inc_RegHsg( hhinc=hhincome_a, hhsize=numprec )
+run;
+
+data hhtype_&year.;
+merge hhtype_1_&year. nonrelatehh_&year. ;
+by serial;
+if hhtype in (4,5,6,7) then do;  /*non family*/
+	if numprec=1 then HHcat=1 ; /*single*/
+end;
+
+if hhtype in (1,2,3) then do; /*family household*/
+    if hhtype=1 & numprec=2 then HHcat=2 ; /*couple without kid*/
+	else HHcar=3; /*other family*/
+end;
+
+if nonrelatehh=1 then HHcat=3; /* non relate households*/
+
+if hhtype in (0, 9) then HHcat=.;
+
+else HHcat=4;
+
+retain HHnumber_&year.=1;
+
+run; 
+
+%mend popbyrace;
+
+%popbyrace(2010);
+%popbyrace(2017);
+
+
+data hhtype_1_2000 (where=(upuma in ("1100101",
+"1100102",
+"1100103",
+"1100104",
+"1100105",
+"2401600",
+"2400300",
+"2401001",
+"2401002",
+"2401003",
+"2401004",
+"2401005",
+"2401006",
+"2401007",
+"2401101",
+"2401102",
+"2401103",
+"2401104",
+"2401105",
+"2401106",
+"2401107",
+"5100101",
+"5100100",
+"5100301",
+"5100302",
+"5100303",
+"5100304",
+"5100305",
+"5100600",
+"5100501",
+"5100502",
+"5100200"
+)));
+set Ipums.Ipums_2000_dc(where=(pernum=1 and gq in (1,2))) Ipums.Ipums_2000_va(where=(pernum=1 and gq in (1,2))) Ipums.ACS_Ipums_2000_md(where=(pernum=1 and gq in (1,2)));
+keep pernum upuma gq Jurisdiction hhwt perwt year serial numprec HHINCOME HHTYPE hud_inc;
+
+  if upuma in ("1100101", "1100102", "1100103", "1100104", "1100105") then Jurisdiction =1;
+  if upuma in ("2401600") then Jurisdiction =2;
+  if upuma in ("2400301", "2400302") then Jurisdiction =3;
+  if upuma in ("2401001", "2401002", "2401003", "2401004", "2401005", "2401006", "2401007") then Jurisdiction =4;
+  if upuma in ("2401101", "2401102", "2401103", "2401104", "2401105", "2401106", "2401107") then Jurisdiction =5;
+  if upuma in ("5101301", "5101302") then Jurisdiction =6;
+  if upuma in ("5159301", "5159302", "5159303", "5159304", "5159305", "5159306", "5159307", "5159308", "5159309") then Jurisdiction =7;
+  if upuma in ("5110701", "5110702" , "5110703") then Jurisdiction =8;
+  if upuma in ("5151244", "5151245", "5151246") then Jurisdiction =9; 
+  if upuma in ("5151255") then Jurisdiction =10; 
+
+	  if hhincome ~=.n or hhincome ~=9999999 then do; 
+		 %dollar_convert( hhincome, hhincome_a, &year., 2017, series=CUUR0000SA0 )
+	   end; 
+  
+	*create HUD_inc - uses 2016 limits but has categories for 120-200% and 200%+ AMI; 
+
+		%Hud_inc_RegHsg( hhinc=hhincome_a, hhsize=numprec )
+run;
+%macro summarizehh(year);
+data hhtype_2000;
+merge hhtype_1_2000 nonrelatehh_2000 ;
+by serial;
+if hhtype in (4,5,6,7) then do;  /*non family*/
+	if numprec=1 then HHcat=1 ; /*single*/
+end;
+
+if hhtype in (1,2,3) then do; /*family household*/
+    if hhtype=1 & numprec=2 then HHcat=2 ; /*couple without kid*/
+	else HHcar=3; /*other family*/
+end;
+
+if nonrelatehh=1 then HHcat=3; /* non relate households*/
+
+if hhtype in (0, 9) then HHcat=.;
+
+else HHcat=4;
+
+retain HHnumber_2000 =1;
```

Comment:
The merge is not working correctly here because the datasets both need to be sorted by the merge variable. I'm also not sure what the retain code is supposed to do?